cockpitlook.cpp

```diff
@@ -1838,25 +1842,26 @@ int PlayerCameraUpdateHook(int* params)
 	//log_debug("Running hooked PlayerCameraUpdate");
 	
 	int playerIndex = params[0];
-	// Apply the offset to the cockpit position relative to the craft OBJ
-	PlayerDataTable[playerIndex].Camera.ShakeX = (g_fXWAUnitsToMetersScale * g_headPos.x);
-	PlayerDataTable[playerIndex].Camera.ShakeY = (g_fXWAUnitsToMetersScale * g_headPos.y);
-	PlayerDataTable[playerIndex].Camera.ShakeZ = (g_fXWAUnitsToMetersScale * g_headPos.z);
-
-	/*log_debug("g_headPos = {%f,%f,%f}", g_headPos.x, g_headPos.y, g_headPos.z);
-	log_debug("CockpitPositionTransformed = {%f,%f,%f}",
-		PlayerDataTable[playerIndex].CockpitPositionTransformedX,
-		PlayerDataTable[playerIndex].CockpitPositionTransformedY,
-		PlayerDataTable[playerIndex].CockpitPositionTransformedZ);
-	log_debug("Camera.Position = {%f,%f,%f}",
-		PlayerDataTable[playerIndex].Camera.PositionX,
-		PlayerDataTable[playerIndex].Camera.PositionY,
-		PlayerDataTable[playerIndex].Camera.PositionZ);*/
 
 	int(*PlayerCameraUpdate)(int) = (int(*)(int)) 0x004EE820;
 	return PlayerCameraUpdate(params[0]);
 }
 
+int CockpitPositionTransformHook(int* params)
+{
+	//log_debug("Running hooked Vector3Transform in CockpitPositionTransformHook\n");
+	
+	Vector3* vec = (Vector3*)params[0];	
+	
+	// Recommended value for g_fXWAUnitsToMetersScale = 25.
+	vec->x += (g_fXWAUnitsToMetersScale * g_headPos.x);
+	vec->z += (g_fXWAUnitsToMetersScale * g_headPos.y);
+	vec->y -= (g_fXWAUnitsToMetersScale * g_headPos.z);
+	void* (*Vector3Transform)(void*, void*) = (void* (*)(void*, void*)) 0x439B30;

```

Comment:
This looks like it can be moved outside this function and made a global function pointer.